contrib/google.golang.org/grpc/option.go

```diff
@@ -60,24 +62,35 @@ func defaults(cfg *config) {
 }
 
 func clientDefaults(cfg *config) {
-	cfg.serviceName = namingschema.NewDefaultServiceName(
+	sn := namingschema.NewDefaultServiceName(
 		defaultClientServiceName,
 		namingschema.WithOverrideV0(defaultClientServiceName),
 	).GetName()
+	cfg.serviceName = func() string { return sn }
 	cfg.spanName = namingschema.NewGRPCClientOp().GetName()
 	defaults(cfg)
 }
 
 func serverDefaults(cfg *config) {
-	cfg.serviceName = namingschema.NewDefaultServiceName(defaultServerServiceName).GetName()
+	// We check for a configured service name, so we don't break users who are accidentally creating their server
+	// before the call `tracer.Start()`
+	if globalconfig.ServiceName() != "" {
+		sn := namingschema.NewDefaultServiceName(defaultServerServiceName).GetName()
+		cfg.serviceName = func() string { return sn }
```

Comment:
I think I'm missing something here.
If there's a global service name, why aren't we using it? e.g. why isn't the code this:
```
if sn := globalconfig.ServiceName(); sn != "" {
	cfg.serviceName = sn
} else { ...
```